src/main/java/uk/nhs/hee/tis/trainee/sync/event/CurriculumMembershipEventListener.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2021 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.sync.event;
+
+import java.util.Optional;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener;
+import org.springframework.data.mongodb.core.mapping.event.AfterDeleteEvent;
+import org.springframework.data.mongodb.core.mapping.event.AfterSaveEvent;
+import org.springframework.data.mongodb.core.mapping.event.BeforeDeleteEvent;
+import org.springframework.stereotype.Component;
+import uk.nhs.hee.tis.trainee.sync.facade.CurriculumMembershipEnricherFacade;
+import uk.nhs.hee.tis.trainee.sync.model.CurriculumMembership;
+import uk.nhs.hee.tis.trainee.sync.model.ProgrammeMembership;
+import uk.nhs.hee.tis.trainee.sync.service.CurriculumMembershipSyncService;
+
+@Component
+public class CurriculumMembershipEventListener
+    extends AbstractMongoEventListener<CurriculumMembership> {
+
+  private final CurriculumMembershipEnricherFacade curriculumMembershipEnricher;
+
+  private CurriculumMembershipSyncService curriculumMembershipSyncService;
+
+  private Cache curriculumMembershipCache;
+
+  CurriculumMembershipEventListener(CurriculumMembershipEnricherFacade curriculumMembershipEnricher,
+                                    CurriculumMembershipSyncService curriculumMembershipSyncService,
+                                   CacheManager cacheManager) {
+    this.curriculumMembershipEnricher = curriculumMembershipEnricher;
+    this.curriculumMembershipSyncService = curriculumMembershipSyncService;
+    curriculumMembershipCache = cacheManager.getCache(ProgrammeMembership.ENTITY_NAME);
```

Comment:
Should we use the cache of CurriculumMembership?

Reply:
Yes indeed! :)